File: Programmers/2/131704. 택배상자/택배상자.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> order) {
    int answer = 0;
    stack<int> st;
    int cur = 1;
    
    for (int o : order)
    {
        while (cur <= o)
        {
            st.push(cur);
            cur++;
        }
        
        if (!st.empty() && st.top() == o)
        {
            st.pop();
            answer++;
        }
        else
        {
            break;
        }
    }
    
    return answer;
}

File: Programmers/2/131704. 택배상자/택배상자.java
import java.util.*;

class Solution {
    public int solution(int[] order) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        int current = 1;
        
        for (int o : order) {
            while (current <= o) {
                stack.push(current);
                current++;
            }
            
            if (!stack.isEmpty() && stack.peek() == o) {
                stack.pop();
                answer++;
            } else {
                break;
            }
        }
        
        return answer;
    }
}